internal/cmd/base/option.go

```diff
@@ -188,3 +190,10 @@ func WithEventGating(with bool) Option {
 		o.withEventGating = with
 	}
 }
+
+// WithPrometheusRegisterer uses the provided prometheus registerer
+func WithPrometheusRegisterer(with prometheus.Registerer) Option {
```

Comment:
I like this approach.  Can you also please add a test to /internal/cmd/base/option_test.go ?